utilities.cpp

```diff
@@ -66,3 +66,46 @@ bool are_from_same_distribution(const std::vector<double> &lhs, const std::vecto
   else
     return false;
 }
+
+bool are_same_distribution(std::uniform_real_distribution<double>lhs,  std::uniform_real_distribution<double> rhs)
+{
+  std::vector<double> lhs_cues(1000);
+  std::vector<double> rhs_cues(1000);
+  std::mt19937_64 rng1;
+  std::mt19937_64 rng2;
+
+  int repeats = 1000;
+  for(int i = 0; i != repeats; i++)
+  {
+      lhs_cues[i] = lhs(rng1);
+      rhs_cues[i] = rhs(rng2);
+  }
+
+  return lhs_cues == rhs_cues;
+}
+
+bool are_same_env_functions(const std::function<double(std::vector<double>)> &lhs,
+                            const std::function<double(std::vector<double>)> &rhs)
+{
+  std::vector<std::vector<double>> input_series(1000);
+
+  for(int i = 0; i != 1000; ++i)
+    {
+      double j = i;
+      std::vector<double> input{j+1, j+2, j+3};
+      input_series[i] = input;
```

Comment:
I think you do not need to decalre j here, j and i are the same and stay the same in this loop. So by not declaring j you just make the code sliker and with less potential sources of errors, plus I would use the `push_back()` here as well
```suggestion
     input_series.pushback({i+1, i+2, i+3});
```

Reply:
I tried that before and it did not like it because i is not a double. I could do static casts instead of making a j but that was less slick rather than more, I figured?